SPECS-CROSS/kernel-headers_cross/kernel-headers_cross.spec

```diff
@@ -0,0 +1,178 @@
+# Globals which should be in a macro file.
+# These should be set programatically in the future.
+%global _host_arch      x86_64
+%global _target_arch    aarch64
+%global _tuple          %{_target_arch}-%{_vendor}-linux-gnu
+%global _cross_name     %{_target_arch}-%{_vendor}-linux-gnu
+
+# Folders which should be in our macro file
+%global _opt                /opt/
+%global _crossdir           /opt/cross/
+
+# Generally we include '/usr' in most paths.
+# Can we also use '/usr' for our paths? This will bring us in line with the
+# %%configure macro which sets these.
+%global _bindir            /bin
+%global _sbindir           /sbin
+%global _libdir            /lib
+%global _lib64dir          /lib64
+%global _libexecdir        /libexec
+%global _datadir           /share
+%global _docdir            /share/doc
+%global _includedir        /include
+%global _infodir           /share/info
+%global _mandir            /share/man
+%global _oldincludedir     /include
+
+
+# Why is this wrong? We get "x86_64-pc-linux-gnu" when eval'd, but our 
+# tools select "aarch64-linux-gnu"
+%global _host_vendor        %{nil}
+
+# If we want our cross compile aware packges to also support native, we
+# need logic to switch modes something like this:
+%if %{_target_arch} != %{_host_arch}
+%global _cross_prefix       %{_crossdir}%{_tuple}/
+%global _cross_sysroot      %{_crossdir}%{_tuple}/sysroot/
+%global _cross_includedir   /usr/%{_host}/%{_tuple}/include/
+%global _cross_infodir      %{_crossdir}%{_tuple}/share/info
+%global _cross_bindir       %{_tuple}/bin
+%global _cross_libdir       %{_tuple}/lib
+%global _tuple_name         %{_tuple}-
+%else
+%global _cross_prefix       %{nil}
+%global _cross_sysroot      %{nil}
+%global _cross_includedir   %{_includedir}
+%global _cross_infodir      %{_infodir}
+%global _cross_bindir       %{_bindir}
+%global _cross_libdir       %{_libdir}
+%global _tuple_name         %{nil}
+%endif
+
+Summary:        Linux API header files
+Name:           %{_cross_name}-kernel-headers
+Version:        5.4.51
+Release:        13%{?dist}
+License:        GPLv2
+URL:            https://github.com/microsoft/WSL2-Linux-Kernel
+Group:          System Environment/Kernel
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Source0:        https://github.com/microsoft/WSL2-Linux-Kernel/archive/linux-msft-%{version}.tar.gz
+BuildArch:      noarch
+Provides:       %{_cross_name}-glibc-kernheaders = %{version}-%{release}
+%description
+The Linux API Headers expose the kernel's API for use by Glibc.
+
+%prep
+%setup -q -n WSL2-Linux-Kernel-linux-msft-%{version}
+
+%build
+make mrproper
+make headers_check
```

Comment:
Do we need to set the `ARCH` variable to the cross target when running this command?

Reply:
Not sure actually, can't hurt to add it though. (I didn't reference the script for these, just did a minimal modification of the existing spec)